st2common/st2common/services/trace.py

```diff
@@ -54,7 +54,8 @@ def _get_valid_trace_context(trace_context):
     """
     Check if tarce_context is a valid type and returns a TraceContext object.
     """
-    assert isinstance(trace_context, (TraceContext, dict))
+    if not isinstance(trace_context, (TraceContext, dict)):
+        raise TypeError("The trace context has a value that is not a dictionary.")
```

Comment:
This exception message should be improved to indicate that `trace_context` can be either a `TraceContext` object or a dictionary.